build/teamcity/cockroach/nightlies/roachtest_nightly_impl.sh

```diff
@@ -62,6 +62,7 @@ fi
 build/teamcity-roachtest-invoke.sh \
   --metamorphic-encryption-probability=0.5 \
   --metamorphic-arm64-probability="${ARM_PROBABILITY:-0.5}" \
+  --metamorphic-cockroach-ea-probability="${COCKROACH_EA_PROBABILITY:-0.2}" \
```

Comment:
Should we also enable AWS and Azure, or do you prefer to do this in stages?

Reply:
Hmm, I'm actually a bit confused. My intention was to enable for all, but I realize now I didn't pass this env var as a docker arg in the cloud specific scripts. I've done so now, but in my TC testing, I was definitely able to get it to enable runtime assertions all the time using the flag 🤔 

Will need to do some more investigating.

Reply:
Yeah that's what i'd think too but looking at the artifacts of the 0.5 select prob run I get

> grep -ri "Runtime assertions enabled" | wc -l
161
> grep -ri "Runtime assertions disabled" | wc -l
15

The tests that are disabled check out with the tests that we opted out last time so it seems like it's getting 100%

I thought maybe the condition is just wrong but:

```
if rand.Float64() < assertionsEnabledProbability {
  enable assertions
}
```
seems correct to me and running it a bunch locally seems like its correct.

Additionally, the 0.6 prob run i just kicked off appears to be doing it correctly so not sure what's going on 😕

Reply:
I ran a seperate branch on 0.375 probability without the docker args [here](https://teamcity.cockroachdb.com/repository/download/Cockroach_Nightlies_RoachtestNightlyGceBazel/16553921:id/acceptance/gossip/peerings/run_1/artifacts.zip!/test.log).

>2024/08/20 19:34:03 test_impl.go:171: rng: 0.697556, flag: 0.375000
>2024/08/20 19:34:03 test_impl.go:183: Runtime assertions disabled

It gets the flag so I think I'm just misunderstanding and we don't actually need the docker args.

Reply:
> but I realize now I didn't pass this env var as a docker arg in the cloud specific scripts. I've done so now, but in my TC testing, I was definitely able to get it to enable runtime assertions all the time using the flag

I guess it doesn't matter much now, but for the record, the reason this works is that we pass `COCKROACH_` prefixed vars to docker automatically to facilitate ad-hoc debugging (if you want to run a test while tweaking some env-var controller functionality, for example).

https://github.com/cockroachdb/cockroach/blob/master/build/teamcity-bazel-support.sh#L10-L22

Reply:
Thank you! Mystery solved 😄